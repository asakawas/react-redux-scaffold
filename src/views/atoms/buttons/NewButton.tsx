import React, { FunctionComponent } from 'react';
import SpacingButton, { SpacingButtonProps } from '~/views/atoms/buttons/SpacingButton';

/**
 * The type of props of NewButton.
 */
export type NewButtonProps = Readonly<
  Pick<SpacingButtonProps, 'ml' | 'disabled'> & {
    /** The event handler called when the button is clicked */
    onNewButtonClick: SpacingButtonProps['onClick'];
  }
>;

const NewButton: FunctionComponent<NewButtonProps> = ({ onNewButtonClick, ml, disabled }) => (
  <SpacingButton
    variant="contained"
    aria-label="new button"
    onClick={onNewButtonClick}
    ml={ml}
    disabled={disabled}
  >
    New
  </SpacingButton>
);

export default NewButton;
