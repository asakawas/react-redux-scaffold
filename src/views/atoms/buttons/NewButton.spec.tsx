import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewButton from './NewButton';

describe('NewButton', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(
      <NewButton onNewButtonClick={() => {}} ml={1} disabled={false} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls the passed handler when it's clicked if not disabled", () => {
    const handler = jest.fn();
    const renderResult = render(<NewButton onNewButtonClick={handler} ml={1} disabled={false} />);

    const tree = renderResult.getByText('New');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(1);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(2);
  });

  test("calls the passed handler when it's clicked if disabled", () => {
    const handler = jest.fn();
    const renderResult = render(<NewButton onNewButtonClick={handler} ml={1} disabled />);

    const tree = renderResult.getByText('New');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(0);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(0);
  });

  test('has an attribute "ml" with its value set to the given number', () => {
    const renderResult = render(<NewButton onNewButtonClick={() => {}} ml={11} disabled={false} />);

    const btn = renderResult.getByLabelText('new button');
    expect(btn).toHaveAttribute('ml', '11');
  });
});
