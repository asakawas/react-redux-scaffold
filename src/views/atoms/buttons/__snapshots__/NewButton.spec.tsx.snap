// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewButton is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-left: 8px;
}

<button
    aria-label="new button"
    class="MuiButtonBase-root MuiButton-root MuiButton-contained c0"
    ml="1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      New
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
